app/_src/kubernetes-ingress-controller/guides/using-ingress-with-grpc.md

```diff
@@ -86,3 +88,182 @@ talks to the upstream service:
     ```bash
     grpcurl -v -d '{"greeting": "Kong Hello world!"}' -insecure $PROXY_IP:443 hello.HelloService.SayHello
     ```
+{% endif_version %}
+
+{% if_version gte:2.9.x %}
+## Overview
+
+This guide walks through deploying a simple [Service][svc] that listens for
+[gRPC connections][gRPC] and exposes this service outside of the cluster using
+{{site.base_gateway}}.
+
+For this example, you will:
+* Deploy a gRPC test application.
+* Route gRPC traffic to it using Ingress or GRPCRoute.
+
+{% include_cached /md/kic/installation.md kong_version=page.kong_version %}
+
+{% include_cached /md/kic/class.md kong_version=page.kong_version %}
+
+[svc]:https://kubernetes.io/docs/concepts/services-networking/service/
+[gRPC]:https://grpc.io/
+
+## Prerequisite
+
+To make `gRPC` requests, you need a client that can invoke gRPC requests.
+In this guide, we use
+[`grpcurl`](https://github.com/fullstorydev/grpcurl#installation).
+Ensure that you have it installed on your local system.
+
+## Deploy a gRPC test application
+
+Add a gRPC deployment and service:
+
+```bash
+echo "---
+apiVersion: v1
+kind: Service
+metadata:
+  name: grpcbin
+  labels:
+    app: grpcbin
+spec:
+  ports:
+  - name: grpc
+    port: 443
+    targetPort: 9001
+  selector:
+    app: grpcbin
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: grpcbin
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: grpcbin
+  template:
+    metadata:
+      labels:
+        app: grpcbin
+    spec:
+      containers:
+      - image: moul/grpcbin
```

Comment:
Hmm now that I look at it https://github.com/moul/grpcbin seems a bit dead with last release from May 2021: https://github.com/moul/grpcbin/releases/tag/v1.0.8 (which does not have its container image https://hub.docker.com/r/moul/grpcbin)

We have a fork for that https://github.com/Kong/grpcbin but no images published (beyond a single `latest` tag: https://hub.docker.com/r/kong/grpcbin/tags 🤔 

I've also tried to use the SHA that's behind `latest` (`ba07ee0f8b44` - https://hub.docker.com/layers/moul/grpcbin/latest/images/sha256-bd8f2ffdd02d0849fad2d1c754eff4402c867e7a3e0552b8992f4590f5687d20?context=explore) but I got `ErrImagePull`.

I guess we can leave this for now then? And perhaps we can push images for https://github.com/Kong/grpcbin to dockerhub so that we can use them.

cc: @mlavacca

Reply:
👍

We can create an issue for this and ask around if there's anyone maintaining our fork at Kong

Reply:
https://github.com/Kong/kubernetes-ingress-controller/issues/3659